diff -rup duc-1.3.3/src/duc/cmd-cgi.c duc-1.3.3-no-header/src/duc/cmd-cgi.c
--- duc-1.3.3/src/duc/cmd-cgi.c	2015-06-22 14:35:07.410366583 -0400
+++ duc-1.3.3-no-header/src/duc/cmd-cgi.c	2015-06-22 14:50:20.558786691 -0400
@@ -26,6 +26,7 @@ static int opt_apparent = 0;
 static char *opt_css_url = NULL;
 static char *opt_database = NULL;
 static int opt_bytes = 0;
+static int opt_no_header = 0;
 static int opt_list = 0;
 static int opt_size = 800;
 static double opt_fuzz = 0.7;
@@ -238,13 +239,15 @@ static void do_index(duc *duc, duc_graph
 	struct duc_index_report *report;
 	int i = 0;
 
-	printf(
-		"Content-Type: text/html\n"
-		"\n"
-		"<!DOCTYPE html>\n"
-		"<head>\n"
-		"  <meta charset=\"utf-8\" />\n"
-	);
+	if(!opt_no_header) {
+		printf(
+			"Content-Type: text/html\n"
+			"\n"
+			"<!DOCTYPE html>\n"
+			"<head>\n"
+			"  <meta charset=\"utf-8\" />\n"
+		);
+	}
 
 	if(opt_css_url) {
 		printf("<link rel=\"stylesheet\" type=\"text/css\" href=\"%s\">\n", opt_css_url);
@@ -256,10 +259,11 @@ static void do_index(duc *duc, duc_graph
 		print_script(path);
 	}
 
-	printf(
-		"</head>\n"
-		"<div id=main>\n"
-	);
+	if(!opt_no_header) {
+		printf("</head>\n");
+	}
+
+	printf("<div id=main>\n");
 
 	printf("<div id=index>");
 	printf(" <table>\n");
@@ -471,6 +475,7 @@ static struct ducrc_option options[] = {
 	{ &opt_apparent,  "apparent",  'a', DUCRC_TYPE_BOOL,   "Show apparent instead of actual file size" },
 	{ &opt_bytes,     "bytes",     'b', DUCRC_TYPE_BOOL,   "show file size in exact number of bytes" },
 	{ &opt_css_url,   "css-url",     0, DUCRC_TYPE_STRING, "url of CSS style sheet to use instead of default CSS" },
+	{ &opt_no_header, "no-header",   0, DUCRC_TYPE_BOOL,   "skip header output" },
 	{ &opt_database,  "database",  'd', DUCRC_TYPE_STRING, "select database file to use [~/.duc.db]" },
 	{ &opt_fuzz,      "fuzz",       0,  DUCRC_TYPE_DOUBLE, "use radius fuzz factor when drawing graph [0.7]" },
 	{ &opt_levels,    "levels",    'l', DUCRC_TYPE_INT,    "draw up to ARG levels deep [4]" },
